Marlin/src/pins/stm32/pins_FYSETC_S6.h

```diff
@@ -0,0 +1,240 @@
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#ifndef STM32F4
+  #error "Oops! Select an STM32F4 board in 'Tools > Board.'"
+#elif HOTENDS > 3 || E_STEPPERS > 3
+  #error "RUMBA32 supports up to 3 hotends / E-steppers."
+#endif
+
+#ifndef BOARD_INFO_NAME
+  #define BOARD_INFO_NAME "FYSETC_S6"
+#endif
+#ifndef DEFAULT_MACHINE_NAME
+  #define DEFAULT_MACHINE_NAME BOARD_INFO_NAME
+#endif
+
+// change the prio to 3 , 2 is for software serial
+//#define TEMP_TIMER_IRQ_PRIO 3
+
+//
+// EEPROM Emulation
+// 
+#define FLASH_EEPROM_EMULATION
+//#define SRAM_EEPROM_EMULATION
+
+//
+// Limit Switches
+//
+#define X_MIN_PIN          PB14
+#define X_MAX_PIN          PA1
+#define Y_MIN_PIN          PB13
+#define Y_MAX_PIN          PA2
+#define Z_MIN_PIN          PA0
+#define Z_MAX_PIN          PA3
+
+//
+// Servos
+// share with Z_MAX_PIN
+//
+#define SERVO0_PIN         PA3
+
+//
+// Steppers
+//
+#define X_STEP_PIN         PE11
+#define X_DIR_PIN          PE10
+#define X_ENABLE_PIN       PE12
+#define X_CS_PIN           PE7
+
+#define Y_STEP_PIN         PD8
+#define Y_DIR_PIN          PB12
+#define Y_ENABLE_PIN       PD9
+#define Y_CS_PIN           PE15
+
+#define Z_STEP_PIN         PD14
+#define Z_DIR_PIN          PD13
+#define Z_ENABLE_PIN       PD15
+#define Z_CS_PIN           PD10
+
+#define E0_STEP_PIN        PD5
+#define E0_DIR_PIN         PD6
+#define E0_ENABLE_PIN      PD4
+#define E0_CS_PIN          PD7
+
+#define E1_STEP_PIN        PE6
+#define E1_DIR_PIN         PC13
+#define E1_ENABLE_PIN      PE5
+#define E1_CS_PIN          PC14
+
+#define E2_STEP_PIN        PE2
+#define E2_DIR_PIN         PE4
+#define E2_ENABLE_PIN      PE3
+#define E2_CS_PIN          PC15
+
+#if HAS_TMC220x
+  /**
+   * TMC2208/TMC2209 stepper drivers
+   *
+   */
+
+  //
+  // Software serial
+  //
+  #define X_SERIAL_TX_PIN  PE9
+  #define X_SERIAL_RX_PIN  PE8
+
+  #define Y_SERIAL_TX_PIN  PE14
+  #define Y_SERIAL_RX_PIN  PE13
+
+  #define Z_SERIAL_TX_PIN  PD11
+  #define Z_SERIAL_RX_PIN  PD12
+
+  #define E0_SERIAL_TX_PIN PD3
+  #define E0_SERIAL_RX_PIN PA15
+
+  #define E1_SERIAL_TX_PIN PC4
+  #define E1_SERIAL_RX_PIN PC5
+
+  #define E2_SERIAL_TX_PIN PE1
+  #define E2_SERIAL_RX_PIN PE0
+#endif
```

Comment:
This needs to define `TMC_BAUD_RATE` for SoftwareSerial.

It is redundant when using SoftwareSerialM, but will be needed by the SoftwareSerial added in #15655, and the one we will use from Arduino_Core_STM32 1.8.

Reply:
This particular change should be made, even if this PR is merged first.